backend/src/resolvers.js

```diff
@@ -11,57 +9,38 @@ module.exports = {
     },
     User: {
       async posts(parent, args , { dataSources }) {
-        const posts = await dataSources.postsApi.getPosts();
-        return posts.filter(post => post.author === parent.name);
+        let posts = await dataSources.postsApi.getPosts();
+        return posts.filter(post => post.author === parent.id);
       }
     },
     Post: {
-      author(parent) {
+      async author(parent, args , { dataSources }) {
+        const userById = await dataSources.usersApi.getUserById(parent.author);
         return {
-          name: parent.author
+          name: userById.name
         };
```

Comment:
If you return a name like this then it's not possible to nest your queries 'indefinitely':

``` graphql
{
   posts {
     title
     votes
     author {
       name
       posts {
         title
       }
     }
   }
}
```
as result, the author's posts will be an empty list.
Here you should return to a User instead of the author's name.

```suggestion
return await dataSources.usersApi.getUserById(parent.author);
```

Reply:
That's 100% correct!